docs/source/reference/api-server/api-server-admin-deploy.rst

```diff
@@ -289,10 +175,165 @@ Test the API server by curling the health endpoint:
 
 If all looks good, you can now start using the API server. Refer to :ref:`sky-api-server-connect` to connect your local SkyPilot client to the API server.
 
-Updating the API server
+.. _sky-api-server-configure-credentials:
+
+Optional: Configure cloud accounts
+----------------------------------
+
+Following tabs describe how to configure credentials for different clouds on the API server. All cloud credentials are stored in Kubernetes secrets.
+
+.. note::
+
+    When you configure credentials after the API server is deployed, an API server restart will be automatically triggered to apply the new credentials. Refer to :ref:`sky-api-server-upgrade` for more details about the potential downtime and mitigation.
+
+.. tab-set::
+
+    .. tab-item:: Kubernetes
+        :sync: kubernetes-creds-tab
+
+        By default, SkyPilot API server is granted permissions to use its hosting Kubernetes cluster and will launch tasks in the same namespace as the API server:
+
+        * To disable using the hosting Kubernetes cluster, set ``kubernetesCredentials.useApiServerCluster=false`` in the Helm chart values.
+        * To use a different namespace for tasks, set ``kubernetesCredentials.inclusterNamespace=<namespace>`` in the Helm chart values.
+
+        .. tip::
+
+            The default permissions granted to the API server works out of box. For further hardening, you can refer to :ref:`Setting minimum permissions in helm deployment <minimum-permissions-in-helm>` to understand the permissions and how to customize them.
+
+        To authenticate to other clusters, first create a Kubernetes secret with the kubeconfig file with :ref:`necessary permissions <cloud-permissions-kubernetes>`:
+
+        .. code-block:: bash
+
+            kubectl create secret generic kube-credentials \
+              --namespace $NAMESPACE \
```

Comment:
Do we want to clarify that this namespace and the one below is for the name space on the hosting kubernetes cluster?

Reply:
We assume some variables will be used throughout the doc, so there is a comment when each variable is defined:

```
    # The NAMESPACE variable will be used throughout the guide
    NAMESPACE=skypilot
    kubectl create namespace $NAMESPACE
```